### PR TITLE
The Lesser Migration

### PR DESCRIPTION
**Pull request by AnthonyLedesma**
### Description

The Great Migration POC demonstrated a successful approach for migrating CoBlocks blocks into Core blocks, providing users with a more consistent and unified experience. The Lesser Migration builds on this success by implementing a staged approach, allowing us to gradually introduce this experience starting with a smaller audience.

As the Gutenberg landscape has evolved, it has become clear that Core blocks offer quality user experience, consistent controls, and regular updates. While it is possible to align CoBlocks with Core, it has proven to be a challenging task to back-port features and functionalities. Additionally, migrations to new features and functionality often require deprecating blocks, adding to bloat and load times.

In summary, The Lesser Migration aims to streamline the user experience by migrating select blocks to Core, reducing maintenance and update overhead, and allowing us to focus on unique and user-enhancing features in CoBlocks.

Also requires a Go branch for proper Go-specific styling.
https://github.com/godaddy-wordpress/go/pull/838

### Screenshots

**Alert Block**
migrateAlertBlock

**Author Block**
migrateAuthorBlock

### Types of changes

Major changes intended for CoBlocks 3.0 release.
PHP and JS changes introducing block migrations. 

### How has this been tested?

Manually tested minor CoBlocks versions back to 2.20.0 all able to successfully migrate.
PHPUnit tests in place. Unit tests cover:
- [x] functionality of core migration logic.
- [x] block parser attribute validation 
- [x] e2e test covering JavaScript logic of migration - Handled in Cypress in GHA.

**To test this manually:**
First, you must set up a 'master' branch of CoBlocks locally. You want to create two new posts. One post is to test the Author block and another is to test the Alert block. When running the block editor and 'master' branch you should insert blocks into the post. We should use features of the block as well. For example, we should have a block with custom colors. A block with custom alignment. Block with a custom class name. A Block using any and all features such as block animation or typography controls like bold and italics. 

Once we have a test post for each block, and each test post is filled with configured blocks, we are ready to test!

Next, we need to check out this branch `cb-30-base` and build. After building these changes, the next time we load one of the test posts, it will migrate the CoBlocks blocks into Core blocks.

For reference:
Alert Block -> Paragraph styled as an alert.
Author Block -> Columns block layout as author. 

### Acceptance criteria

- [x] Blocks should migrate. 
- [x] Tests should provide confidence of functionality.

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: .dev/tests/phpunit/includes/block-migrate/test-coblocks-block-migration.php [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+		$block_wrapper->setAccessible( true );
+
+		$parsed_block = parse_blocks(
+			<<<BLOCKHTML
+			<!-- wp:coblocks/test-block /-->
+			BLOCKHTML
+		);
+		$this->instance->migrate( $parsed_block[0]['attrs'], $parsed_block[0]['innerHTML'] );
+
+		// var_dump( $block_wrapper->invoke( $this->instance ) );
```
Comment (kopepasah) [inline .dev/tests/phpunit/includes/block-migrate/test-coblocks-block-migration.php:63]: @AnthonyLedesma lets remove this debug code.

File: includes/block-migrate/loader.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			array( $migration_runner, 'register' ),
+			array(
+				CoBlocks_Alert_Migration::class,
+				CoBlocks_Author_Migration::class,
+			)
+		);
+
+		$parsed_blocks = array_map( array( $migration_runner, 'migrate_block' ), $parsed_blocks );
+
+		// var_dump($parsed_blocks);
```
Comment (olafleur-godaddy) [inline includes/block-migrate/loader.php:50]: commented out code

**Comment by GenieTim** [comment]
At least the migration from alert to paragraph does not work well with UTF-8 characters (e.g., Umlaute as often used on German sites, sucha as ä, ö or ü) and is currently destroying one page after another on my sites, whenever I want to change anything else on the page, leading to much extra work of changing the butchered characters back. The list of changes is unfortunately very extensive in this PR, making it hard for me to find an entry point to submit a fix – could you kindly give an indication on where to start, dear @AnthonyLedesma ?

**Comment by AnthonyLedesma** [comment]
Hi @GenieTim,

I am so sorry to hear about that fact with UTF8 chars. This fix I think has to do with how the content is loaded into PHP and parsed locally. It seems like the content is being imported in a way that does not handle some characters. 

I took a quick look and tested it out with special characters, made some modifications to the migrate logic, and have a PR that is close to ready. I hope to have this fix out soon.

**Comment by AnthonyLedesma** [comment]
PR up to resolve special characters bug.
https://github.com/godaddy-wordpress/coblocks/pull/2526

**Pull request merged**